pages/[chain]/collection/[contract].tsx

```diff
@@ -131,13 +131,16 @@ const CollectionPage: NextPage<Props> = ({ id, ssr }) => {
     (stage) => stage.kind === 'public'
   )
 
-  const mintPrice = mintData?.price?.amount?.decimal
-    ? mintData?.price?.amount?.decimal === 0
-      ? 'Free'
-      : `${
-          mintData?.price?.amount?.decimal
-        } ${mintData?.price?.currency?.symbol?.toUpperCase()}`
-    : undefined
+  const mintPrice =
```

Comment:
think maybe we can cut down the duplication here
```js
const priceDecimal = mintData?.price?.amount?.decimal 
const mintPrice = 
  typeof priceDecimal === 'number' && 
  priceDecimal !== null &&
  priceDecimal !== undefined &&
  etc
  ```
  
  
  
  
  ...
  
  also wonder if we can use lodash a bit here
  
  
  ```js
  const priceDecimal = mintData?.price?.amount?.decimal 
const mintCurrency = mintData?.price?.currency?.symbol?.toUpperCase()
  
const mintPrice = 
  _.isZero(priceDecimal) 
    ? 'Free' : _.isNumeric(priceDecimal)  
      ? `${priceDecimal} ${mintCurrency}`
       : undefined
```